### DISCUSSION TITLE
Saved Image Colors are Inverted.

### DISCUSSION THREAD

**Discussion by izumeee** [natsuk4ze/gal#239] [Q&A] [answered]
### Did you check the documentation and it did not help?

- [X] I have carefully read the [README](https://github.com/natsuk4ze/gal/blob/main/README.md)
- [X] I have carefully searched the [existing documentation](https://github.com/natsuk4ze/gal/wiki)
- [X] I have carefully searched the [existing answer](https://github.com/natsuk4ze/gal/discussions)
- [X] I have carefully read the [contribution guide](https://github.com/natsuk4ze/gal/blob/main/CONTRIBUTING.md)

### Descrpition

When I used the [scroll_screenshot](https://pub.dev/packages/scroll_screenshot) package's `captureAndSaveScreenshot` method to capture a screenshot and obtained the base64String, I executed the following code:
`Gal.putImageBytes(base64Decode(base64String));
`
However, the image colors appeared inverted, as if in dark mode. This behavior was observed on both iOS and Android.
Additionally, I tried using the [image_gallery_saver](https://pub.dev/packages/image_gallery_saver) package to process the same base64String with the following code, but the color inversion did not occur:
`ImageGallerySaver.saveImage(base64Decode(base64String));
`
This was confirmed on iOS (Android has not been tested due to build errors).

When viewing the image saved with putImageBytes in the iOS "Photos" app or Android "Files" app, the thumbnail displays the correct colors, but when viewing the full image, the colors appear inverted.

- This issue occurs with both .jpg and .png formats.
- It is reproducible whether dark mode is enabled or disabled on the device.
- The issue was reproduced on Android 13, 14, iOS 17.5, and 18.0.
- It occurs on both real devices and simulators.
- I also confirmed that the issue occurs with the following code:
`final tempDir = await getTemporaryDirectory();`
`Gal.putImage(tempDir.path);`
I have attached the base64 text file that was passed as an argument, as well as the saved image files for reference.
[base64String_2024-10-24_11-08_21335.txt](https://github.com/user-attachments/files/17501828/base64String_2024-10-24_11-08_21335.txt)
![2024-10-24_11-08_21414](https://github.com/user-attachments/assets/d10d5409-a04f-4cad-a613-4ed15cc7fab3)

**Comment by natsuk4ze** [comment]
Hmm. This is a interesting issue.
Is scroll_screenshot a private repositor?, it is on pub.dev but the repository seems to be 404.

**Reply by izumeee** [reply]
What scroll_screenshot is doing is the same as described in the following article. I actually tested it with the sample code from the article, and the same issue was reproduced.
https://zenn.dev/ohtsuki/articles/ecad2879604e25

**Reply by natsuk4ze** [reply]
I wanted to read the source code of the scroll_screenshot package.
For the time being, to identify the problem, can you convert the appropriate image to a base64 string without using the package and then save it using gal?

**Reply by izumeee** [reply]
Sorry about the confusion! The issue wasn’t with your package after all. It turns out the problem was just that I didn’t include the background in the screenshot. Once I made sure the background was part of the capture area, everything worked perfectly.

Thanks so much for the awesome package, and sorry for the fuss!

**Reply by natsuk4ze** [reply, marked as answer]
@izumeee No problem! Glad to hear your issue has been resolved. Please press "Mark as answer" if you are good to complete the thread.
